### PR TITLE
[connectivity] Use platform_interface in the core plugin

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

An improvement over the current platform_interface implementation, migrating more logic into the method channel platform. 

Now the core plugin makes use of the platform_interface.

## Related Issues

https://github.com/flutter/flutter/issues/46307
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by franciscojma86** [comment]
Mostly @ditman work, but happy to take the credit :)

**Comment by ditman** [comment]
@franciscojma86 you can't push changes to two packages at the same time, because our testing won't know what you do.

You first need to PR and release the changes to the `platform_interface`, and then the changes to the core plugin to use the new platform interface. That way you can use the correct version in the dependency of your core plugin.

**Comment by ditman** [comment]
(If you remove the changes to the core plugin from this PR, it'll become publishable for the platform_interface package)

File: packages/connectivity/connectivity_platform_interface/lib/src/method_channel_connectivity.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    return methodChannel.invokeMethod<String>('wifiIPAddress');
+  }
+
+  @override
+  Future<LocationAuthorizationStatus> requestLocationServiceAuthorization({
+    bool requestAlwaysLocationUsage = false,
+  }) {
+    // `assert(Platform.isIOS)` will prevent us from doing dart side unit testing.
+    // TODO: These should noop for non-Android, instead of throwing, so people don't need to rely on dart:io for this.
+    assert(!Platform.isAndroid);
```
Comment (ditman) [inline packages/connectivity/connectivity_platform_interface/lib/src/method_channel_connectivity.dart]: I thought about this a little bit, and we need to remove these `Platform` asserts, they don't add anything to this plugin (other than a dependency in dart:io).

We can do this in other PR, though.

Comment (franciscojma86) [reply packages/connectivity/connectivity_platform_interface/lib/src/method_channel_connectivity.dart]: Yeah I thought so too, but I think the first changes I'd like to keep the API as similar as possible to the current behavior.

Comment (ditman) [reply packages/connectivity/connectivity_platform_interface/lib/src/method_channel_connectivity.dart]: Yep, sounds good to me!

**Pull request merged**